server/src/main/java/org/elasticsearch/plugins/ActionPlugin.java

```diff
@@ -128,6 +129,17 @@
         return null;
     }
 
+    /**
+     * TODO discuss
+     * only one plugin can override this.
+     * however if we allow some kind of wrapping/composition of factories, then more could override
+     * how about the order then?
+     * factory could be returning a builder so that chaining of factories can add more stuff to it?
+     */
+    default RestRequestFactory getRestRequestFactory() {
```

Comment:
instead of extending action plugin, what if we just created a new interface type ? Single plugins are allowed to extend multiple *Plugin interfaces, and might be a bit cleaner and encapsulted.